devito/ir/support/utils.py

```diff
@@ -81,7 +81,7 @@ def build_iterators(mapper):
         for d, offs in v.items():
             if d.is_Stepping:
                 values = iterators.setdefault(d.parent, [])
-                for i in sorted(offs):
+                for i in sorted(offs, key=lambda x: -float("inf") if x == 0 else x):
```

Comment:
@mloubout - can you leave a comment to explain this piece of black magic.

Reply:
done

Reply:
OK. I'm not 100% sure I like this fix (what if SymPy changes its internal policy for canonical ordering? we should actually sort based on that somehow....), but I can buy it for now

Reply:
After my investigation yesterday (for other reasons), I can say that this ordering

```
(t, t-1, t+1)
```

is happening in SymPy because "Symbols always precede operations such as Add and Mul". So `t` is a Symbol while `t-1` and `t+1` are Add's, hence the ordering above. However, there are big TODOs in SymPy that this might change in the future (they're explicitly talking about dropping their `ordering_of_classes` (u may wanna `git grep ordering_of_classes` in SymPy to see what I mean). 

So just to be sure. When you say: 

```
follows sympy's index ordering (t, t-1, t+1) afer modulo replacement
```

what do you exactly mean?

Reply:
>what do you exactly mean?
I mean  `t0=t, t1=t-1, t2=t+1` because for sympy `t0<t1` and `t1<t2` , while curently it generates it by offset and offset order is {-1, 0, 1} and therefore generate  `t0=t-1, t1=t, t2=t+1` which has not the same order.